python/across_api/burstcube/toorequest.py

```diff
@@ -0,0 +1,437 @@
+        """
+        # Shouldn't run this unless a created_by is set
+        if self.username is None:
+            raise HTTPException(401, "Username not set.")
+
+        # Validate supplied BurstCubeTOO values against the Schema
+        if not self.validate_post():
+            return False
+
+        # Set the created_on and created_by fields
+        self.created_on = Time.now()
+        self.created_by = self.username
+
+        # Set the ID of this TOO
+        self.id = hashlib.md5(
+            f"{self.trigger_time}{self.exposure}{self.offset}".encode()
+        ).hexdigest()
+
+        # Check if this BurstCubeTOO exists. As the id is just a hash of the
+        # trigger_time, exposure and offset, then repeated requests for values
+        # that match this will be caught.
+        try:
+            response = self.table.get_item(Key={"id": self.id})
+            if response.get("Item"):
+                raise HTTPException(409, "BurstCubeTOO already exists.")
+        except botocore.exceptions.ClientError:
+            pass
+
+        # Check for various TOO constraints
+        if not self.check_constraints():
+            self.status = TOOStatus.rejected
+
+        # Write BurstCubeTOO to the database
+        json_too = json.loads(self.schema.model_dump_json(), parse_float=Decimal)
+        self.table.put_item(Item=json_too)
+
+        return True
+
+
+class BurstCubeTOORequests(ACROSSAPIBase):
+    """
+    Class to fetch multiple BurstCubeTOO requests, based on various filters.
+
+    Note that the filtering right now is based on DynamoDB scan, which is not
+    very efficient. This should be replaced with a query at some point.
+
+    Parameters
+    ----------
+    begin
+        Start time of plan search
+    end
+        End time of plan search
+    limit
+        Limit number of searches
+    duration
+        Duration of time to search from now
+
+    Attributes
+    ----------
+    entries
+        List of BurstCubeTOO requests
+    status
+        Status of BurstCubeTOO query
+    """
+
+    _schema = BurstCubeTOORequestsSchema
+    _get_schema = BurstCubeTOORequestsGetSchema
+    mission = "ACROSS"
+
+    def __getitem__(self, i):
+        return self.entries[i]
+
+    def __len__(self):
+        return len(self.entries)
+
+    def __init__(
+        self,
+        begin: Optional[Time] = None,
+        end: Optional[Time] = None,
+        duration: Optional[u.Quantity] = None,
+        limit: Optional[int] = None,
+    ):
+        # Default parameters
+        self.limit = limit
+        self.begin = begin
+        self.end = end
+        self.duration = duration
+        # Attributes
+        self.entries: list = []
+
+        # Parse Arguments
+        if self.validate_get():
+            self.get()
+
+    def get(self) -> bool:
+        """
+        Get a list of BurstCubeTOO requests
+
+        Returns
+        -------
+        bool
+            Did this work? True | False
```

Comment:
Why return `False` on failure? Why not raise an exception?

Reply:
To be fair, it won't ever return `False`. I removed that line.